classes/shop/Shop.php

```diff
@@ -703,6 +703,8 @@ public static function cacheShops($refresh = false)
                 }
 
                 $row = $row + array('theme_name' => '');
+				$domain_with_protocol = (Configuration::get('PS_SSL_ENABLED', null, $row['id_shop_group'], $row['id_shop']) 
```

Comment:
`Tools::getShopProtocol` exists to get you the correct protocol

Reply:
Yes, but only for current shop.

Reply:
Yes, but this get value always for current shop in context.